### PR TITLE
Improved accuracy and detail of pre-populating forms from a URL

### PR DESCRIPTION
**Pull request by gregharvey**
I hope these are an improvement, they're accurate as far as I can tell from my testing.

### CODE REVIEW

**Review by RCheesley — CHANGES_REQUESTED** [review]
@gregharvey sorry I started a review ages ago and for some reason didn't post it!

I've made some suggestions for improvement as inline comments.

It might be helpful to reference the styleguide here: https://docs.mautic.org/en/home/contributing which has some guidelines on things like using links macros. We are trying to update these things as we go, so it would be super appreciated if you could look to pop any links in a block of macros at the bottom of the page.

Shout if you need any assistance!

**Comment by RCheesley** [comment]
@gregharvey could you take a look at the proposed changes so we can get this merged?

File: pages/10.Components/03.Forms/01.Managing Forms/docs.en.md [hunk 1]
```diff
@@ -103,28 +103,73 @@ It is possible to choose a theme for a form. If you do so and the theme supports
 
 ## Pre-populate a form field value
 
-It is possible to pre-populate the value of a form field from the URL query parameters.
+> **ProTip**
+>
+> **When testing the pre-population of forms you should always use a browser in incognito mode, because being logged in as an administrator can confuse Mautic.**
+
+It is possible to pre-populate the value of a form field from the URL query parameters. However, this only works for form fields that are linked to contact fields, all of which can be found on the Settings -> Custom fields page. (In spite of the name, this page lists all fields, including core ones.)
```
[the remark this answers is not in the document]
Comment (gregharvey) [reply pages/10.Components/03.Forms/01.Managing Forms/docs.en.md:110]: "this is configured when you create the form" <- I did this ages ago, so I can't remember any more, but looking at my notes I'm not sure this is the case?

File: pages/10.Components/03.Forms/01.Managing Forms/docs.en.md [hunk 2]
```diff
@@ -103,28 +103,73 @@ It is possible to choose a theme for a form. If you do so and the theme supports
...
 
-So, what happened is `{pagelink=1}` was converted into the landing page URL and had `?ct=A_REALLY_LONG_STRING` appended. The really long string is encoded information about the contact which includes the contact ID. Each `{contactfield=FIELDALIAS}` was replaced with the contact's data. When the contact clicks the link, they will be taken to the landing page with the embedded form, and the form's `email` input will be pre-populated with the value passed through the URL.
+Note, your form field alias has nothing to do with the URL, as long as the form field exists, is linked to the `product` contact field and set to auto-fill as described above, and has the appropriate values, A and B, the parameter name in your querystring will be the contact alias and Mautic will connect the parameter to the matching form field.
 
 ### Remove Contact from Do Not Contact (undo unsubscribe)
```
[the remark this answers is not in the document]
Comment (gregharvey) [reply pages/10.Components/03.Forms/01.Managing Forms/docs.en.md:175]: Added the macro. You want the links to go to GitHub tags, I suppose?

**Comment by gregharvey** [comment]
Sorry, I get so many GitHub notifications they're almost useless when it comes to spotting individual PRs needing a response! :-(

File: pages/10.Components/03.Forms/01.Managing Forms/docs.en.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (RCheesley) [reply pages/10.Components/03.Forms/01.Managing Forms/docs.en.md:175]: Take a look at the docs here for linking to releases: https://docs.mautic.org/en/home/contributing#linking-to-release-versions

**Comment by RCheesley** [comment]
No worries! Have replied to the last comment that is outstanding :)

**Comment by RCheesley** [comment]
@gregharvey this one has a conflict to resolve now. Can you take a look please and also at the outstanding comments? Thanks!

**Pull request closed without merging**